modules/consensus/synchronize.go

```diff
@@ -260,3 +273,96 @@ func (cs *ConsensusSet) relayBlock(conn modules.PeerConn) error {
 	}
 	return nil
 }
+
+// rpcRelayHeader is an RPC that accepts a block header from a peer.
+func (cs *ConsensusSet) rpcRelayHeader(conn modules.PeerConn) error {
+	// Decode the block header from the connection.
+	var h types.BlockHeader
+	err := encoding.ReadObject(conn, &h, types.BlockHeaderSize)
+	if err != nil {
+		return err
+	}
+
+	// Start verification inside of a bolt View tx.
+	cs.mu.RLock()
+	err = cs.db.View(func(tx *bolt.Tx) error {
+		// Do not accept a header if the database is inconsistent.
+		if inconsistencyDetected(tx) {
+			return errInconsistentSet
+		}
+
+		// Do some relatively inexpensive checks to validate the header
+		return cs.validateHeader(boltTxWrapper{tx}, h)
+	})
+	cs.mu.RUnlock()
+	if err == errOrphan {
+		// If the header is an orphan, try to find the parents.
+		//
+		// TODO: log error returned if non-nill? OR we could not use a goroutine and
+		// just return the result of the RPC call. Is it back practice to call an RPC
+		// from inside another RPC?
+		go cs.gateway.RPC(modules.NetAddress(conn.RemoteAddr().String()), "SendBlocks", cs.threadedReceiveBlocks)
+		return nil
+	} else if err != nil {
+		return err
+	}
+	// If the header is valid and extends the heaviest chain, fetch, accept it,
+	// and broadcast it.
```

Comment:
But all of those things are done. I want to make it very clear that the RPC to fetch the corresponding block to a block header will do all of those things.